src/pilot/assign_role.py

```diff
@@ -296,11 +296,15 @@ def define_controller_logical_disks(drac_client, raid_controller_name):
     raid_10_logical_disk = define_single_raid_10_logical_disk(
         drac_client, raid_controller_name)
 
+    # None indicates an error occurred.
     if raid_10_logical_disk is None:
         return None
 
     logical_disks = list()
-    logical_disks.append(raid_10_logical_disk)
+
+    # Add the disk to the list only if it is not empty.
+    if logical_disks:
```

Comment:
This doesn't make sense as logical_disks is always initialized to empty list this code never executes, I think you are supposed to check for  if raid_10_logical_disks: ?

Reply:
Oops..!!..Apologies and corrected the same.